### PR TITLE
added attached_disk initial value

### PR DESCRIPTION
**Pull request by rohan-searce**
Updated main.tf with attached_disks block & added instance_name output

Added one more block in `lifecycle.ignore_changes`, i.e. as it will be required at the time of attaching an additional disk to GCP Virtual Machines.

Added `instance_name` output variable in outputs.tf file. We need this output for naming convention of VM related services like disk management.

I've tested these changes locally, they are working properly.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -62,6 +62,7 @@ resource "google_compute_instance" "vm_instance" {
   metadata = {
     enable-oslogin = (var.os_login_enabled ? "TRUE" : "FALSE") # see https://cloud.google.com/compute/docs/instances/managing-instance-access#enable_oslogin
     windows-keys   = ""                                        # Placeholder to ignore changes. See https://www.terraform.io/docs/configuration/resources.html#ignore_changes
+    attached_disk  = null
```
Comment (syedrakib) [inline main.tf]: Kindly add the helper comment like given for windows-keys. It allows us developers to know `null` is not the intended value of this field but rather just a placeholder value - so in future it helps us avoid looking into this for any potential trouble-shooting efforts

File: outputs.tf [hunk 2]
```diff
@@ -18,3 +18,23 @@ output "sa_roles" {
...
+
+output "instance_id" {
+  description = "The server-assigned unique identifier of this instance. Example: 4567719474035761998"
+  value       = google_compute_instance.vm_instance.instance_id
+}
+
+output "instance_name" {
+  description = "The generated name of the GCloud VM Instance with format {{instance-name}}-vm-{{suffix with 4 characters random string separated with -}}. Example: vm-instance-tf-1a2b"
+  value       = local.instance_name
+}
```
Comment (syedrakib) [inline outputs.tf]: A good practice is to add at least one new-empty-line at the end of source files for various text-processing tools.

Looks like here line 40 itself is the last line. Just add a new-empty-line and that should be good.

**Review by syedrakib — DISMISSED** [review]
Looks good to me :)

Thanks for the changes. Just a few minor nitpicks here.

File: main.tf (hunk 1 above)
Comment (rohan-searce) [reply main.tf]: Done

File: outputs.tf (hunk 2 above)
Comment (rohan-searce) [reply outputs.tf]: Done

**Review by syedrakib — APPROVED** [review]

**Pull request merged**